src/interface/serial_node.py

```diff
@@ -56,6 +56,8 @@ def read_block(self, interface, command, size, max_retries=MAX_RETRY_COUNT, chec
                 try:
                     self.io_request = monotonic()
                     self.serial.flushInput()
+                    gevent.sleep(0.01)  # Brief delay for any in-flight ESP32 boot messages
+                    self.serial.flushInput()  # Second flush to catch delayed boot messages
```

Comment:
Similar to what I said on PR #1082, I am not comfortable adding a 'sleep' into this low-level interface function.  This issue seems like the issue that BOOTLOADER_CHILL_TIME addresses (for Arduino).  We could see about adding a 'flushInput()' to the 'discover()' function, after "node = SerialNode(...)"  Would that fix the issue with 'esp32'?